helpers/decorators/auth.py

```diff
@@ -0,0 +1,131 @@
+import jwt
+from flask import current_app, request, jsonify
+from functools import wraps
+from graphql import GraphQLError
+from helpers.database import db_session
+from sqlalchemy.exc import SQLAlchemyError
+from graphql import GraphQLError
+
+from api.user.models import User
+from api.role.models import Role
+from api.user_role.models import UsersRole
+
+
+class Authentication:
+    """ Authenicate token and all role based authorization
+    :methods
+        verify
+        decode_token
+        user_roles
+        auth_required
+    """
+
+    def verify(self, token):
+        """
+        verify token
+
+        :params
+            token
+        return:
+            func|string
+        """
+        if token:
+            return self.decode_token(token)
+        return jsonify({
+            'message': 'This endpoint requires you to be authenticated.'}), 401
+
+    def decode_token(self, auth_token):
+        """
+        Decodes the auth token
+        :param
+            auth_token
+        :return
+            integer|string
+        """
+        SECRET_KEY = current_app.config['SECRET_KEY']
+
+        try:
+            payload = jwt.decode(auth_token, SECRET_KEY)
+            return payload
+        except jwt.ExpiredSignatureError:
+            return jsonify({
+                'message': 'Signature expired. Please log in again.'}), 401
+        except jwt.InvalidTokenError:
+            return jsonify({
+                'message': 'Invalid token. Please Provied a valid token!'
```

Comment:
You have typo **Provied**, I think you meant **Provide**

Reply:
Thank you.